src/rdkafka_cgrp.c

```diff
@@ -2807,17 +2872,34 @@ void rd_kafka_cgrp_handle_ConsumerGroupHeartbeat(rd_kafka_t *rk,
                                     sizeof(assigned_topic_partitions_str), 0);
                         }
 
-                        rd_rkb_dbg(rkb, CGRP, "HEARTBEAT",
-                                   "Heartbeat response received target "
-                                   "assignment \"%s\"",
-                                   assigned_topic_partitions_str);
+                        rd_kafka_dbg(
+                            rk, CGRP, "HEARTBEAT",
+                            "ConsumerGroupHeartbeat response received target "
+                            "assignment \"%s\"",
+                            assigned_topic_partitions_str);
                 }
 
                 if (assigned_topic_partitions) {
                         RD_IF_FREE(rkcg->rkcg_next_target_assignment,
                                    rd_kafka_topic_partition_list_destroy);
-                        rkcg->rkcg_next_target_assignment =
-                            assigned_topic_partitions;
+                        rkcg->rkcg_next_target_assignment = NULL;
+                        if (rd_kafka_cgrp_consumer_is_new_assignment_different(
+                                rkcg, assigned_topic_partitions)) {
+                                /* We don't update the next_target_assignment
```

Comment:
Move this comment as rd_kafka_cgrp_consumer_is_new_assignment_different documentation